qiskit_nature/second_q/operators/fermionic_op.py

```diff
@@ -411,9 +412,12 @@ def mul(self, other: complex) -> FermionicOp:
 
     def compose(self, other: FermionicOp) -> FermionicOp:
         if not isinstance(other, FermionicOp):
-            raise TypeError(
-                f"Unsupported operand type(s) for *: 'FermionicOp' and '{type(other).__name__}'"
-            )
+            from .mixed_op import MixedOp
+            return MixedOp(([self, other], 1))
```

Comment:
I'm not sure how we could do this for the case `FermionicOp @ SpinOp`, because there isn't any `MixedOp` involved yet, so `MixedOp.__rmatmul__ `  would not be called, right?